x/dex/cache/cache.go

```diff
@@ -18,225 +18,158 @@ const (
+			for _, blockOrder := range *blockOrders {
+				copy.GetBlockOrders(contractAddr, pair).AddOrder(blockOrder)
 			}
 		}
-		newMarketBuys := append(o.MarketBuys, order)
-		if idx != -1 {
-			copy(newMarketBuys[idx+1:], newMarketBuys[idx:])
-			newMarketBuys[idx] = order
-		}
-		o.MarketBuys = newMarketBuys
-	case types.PositionDirection_SHORT:
-		for i, existingOrder := range o.MarketSells {
-			if existingOrder.WorstPrice.GT(order.WorstPrice) {
-				idx = i
-				break
+	}
+	for contractAddr, _map := range s.BlockCancels {
+		for pair, blockCancels := range _map {
+			for _, blockCancel := range *blockCancels {
+				copy.GetBlockCancels(contractAddr, pair).AddOrderIdToCancel(blockCancel.Id, blockCancel.Initiator)
 			}
 		}
-		newMarketSells := append(o.MarketSells, order)
-		if idx != -1 {
-			copy(newMarketSells[idx+1:], newMarketSells[idx:])
-			newMarketSells[idx] = order
-		}
-		o.MarketSells = newMarketSells
-	default:
-		panic("Unknown direction")
-	}
-}
-
-func (o *Orders) AddLimitOrder(order LimitOrder) {
-	switch order.Direction {
-	case types.PositionDirection_LONG:
-		o.LimitBuys = append(o.LimitBuys, order)
-	case types.PositionDirection_SHORT:
-		o.LimitSells = append(o.LimitSells, order)
-	default:
-		panic("Unknown direction")
 	}
-}
-
-func (o *Orders) AddCancelOrder(order CancelOrder) {
-	switch order.Direction {
-	case types.PositionDirection_LONG:
-		o.CancelBuys = append(o.CancelBuys, order)
-	case types.PositionDirection_SHORT:
-		o.CancelSells = append(o.CancelSells, order)
-	default:
-		panic("Unknown direction")
+	for contractAddr, deposits := range s.DepositInfo {
+		for _, deposit := range *deposits {
+			copy.GetDepositInfo(contractAddr).AddDeposit(deposit)
+		}
 	}
-}
-
-func (o Orders) String() string {
-	return fmt.Sprintf(
-		"Limit Buys: %d, Limit Sells: %d, Market Buys: %d, Market Sells: %d, Cancel Buys: %d, Cancel Sells: %d, Cancel Alls: %d",
-		len(o.LimitBuys), len(o.LimitSells),
-		len(o.MarketBuys), len(o.MarketSells),
-		len(o.CancelBuys), len(o.CancelSells),
-		len(o.CancelAlls),
-	)
-}
-
-func NewOrders() *Orders {
-	return &Orders{
-		LimitBuys:   []LimitOrder{},
-		LimitSells:  []LimitOrder{},
-		MarketBuys:  []MarketOrder{},
-		MarketSells: []MarketOrder{},
-		CancelBuys:  []CancelOrder{},
-		CancelSells: []CancelOrder{},
-		CancelAlls:  []CancelAll{},
+	for contractAddr, liquidations := range s.LiquidationRequests {
+		for _, liquidation := range *liquidations {
+			copy.GetLiquidationRequests(contractAddr).AddNewLiquidationRequest(liquidation.Requestor, liquidation.AccountToLiquidate)
+		}
 	}
+	return copy
 }
 
-type OrderPlacement struct {
-	Id         uint64
-	Price      sdk.Dec
-	Quantity   sdk.Dec
-	Creator    string
-	OrderType  types.OrderType
-	Direction  types.PositionDirection
-	Effect     types.PositionEffect
-	PriceDenom string
-	AssetDenom string
-	Leverage   sdk.Dec
-}
+// All new orders attempted to be placed in the current block
+type BlockOrders []types.Order
```

Comment:
can we move this to the top of the `MemState` definition

Reply:
:+1: